java/src/main/java/com/powsybl/python/PyPowsyblNetworkApiLib.java

```diff
@@ -398,6 +400,21 @@ public static void updateNetworkElementsWithSeries(IsolateThread thread, ObjectH
         });
     }
 
+    @CEntryPoint(name = "removeNetworkElements")
+    public static void removeNetworkElements(IsolateThread thread, ObjectHandle networkHandle, CCharPointerPointer cElementIds,
+                                             int elementCount, PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+        doCatch(exceptionHandlerPtr, () -> {
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+            List<String> elementIds = CTypeUtil.toStringList(cElementIds, elementCount);
+            elementIds.forEach(elementId -> {
+                List<Connectable> connectables = network.getConnectableStream()
+                        .filter(connectable -> connectable.getId().equals(elementId))
```

Comment:
As you have suggested, we should not limit ourselves to connectables.

Trying to define the list of other identifiables we want to remove:
 - substations
 - voltage levels
 - switches
 - HVDC lines